src/main/scala/fable/Config.scala

```diff
@@ -152,7 +149,100 @@ object Config {
                  Deserializer[V].instance.getClass.getName)
     }
 
-    private def addTrustStoreProperties[F[_]: Sync](
+  }
+
+  object Consumer {
+    sealed trait AutoOffsetReset {
+      def renderString: String
+    }
+
+    object AutoOffsetReset {
+      case object Earliest extends AutoOffsetReset {
+        val renderString = "earliest"
+      }
+
+      case object Latest extends AutoOffsetReset {
+        val renderString = "latest"
+      }
+
+      case object None extends AutoOffsetReset {
+        val renderString = "none"
+      }
+
+      implicit val autoOffsetResetReader: ConfigReader[AutoOffsetReset] =
+        pureconfig.ConfigReader[String].emap { string =>
+          List(Earliest, Latest, None)
+            .filter(_.renderString == string)
+            .headOption
+            .toRight(pureconfig.error
+              .CannotConvert(string, "auto offset", "Unkown value"))
+        }
+    }
+
+    implicit val consumerConfigReader: ConfigReader[Consumer] = deriveReader
+  }
+
+  /**
+    * Configuration options for constructor a Kafka [[Producer]].
+    *
+    * @see
+    * [[https://kafka.apache.org/21/javadoc/org/apache/kafka/clients/producer/ProducerConfig.html
+    * ProducerConfig]] for details on producer configuration.
+    *
+    * @constructor
+    * @param clientCertificate SSL certificate used to authenticate the client
+    * @param clientKey SSL private key used to authenticate the client
+    * @param sslIdentificationAlgorithm the algorithm used to identify the
+    * server hostname
+    * @param trustedCertificate SSL certificate used to authenticate the server
+    * @param uris bootstrap URIs used to connect to a Kafka cluster.
+    */
+  case class Producer(
+      clientCertificate: Option[String],
+      clientKey: Option[String],
+      sslIdentificationAlgorithm: Option[String],
+      trustedCertificate: Option[String],
+      uris: URIList
+  ) extends Common(clientCertificate,
+                     clientKey,
+                     sslIdentificationAlgorithm,
+                     trustedCertificate,
+                     uris) {
+    def properties[F[_]: Sync, K: Serializer, V: Serializer]: F[Properties] = {
+      val result = new Properties()
+      addCommonProperties(result)
+      addSerializationProperties[K, V](result)
+
+      addTrustStoreProperties[F](result) *>
+        addClientStoreProperties[F](result) *>
+        Sync[F].pure(result)
+    }
+
+    private def addSerializationProperties[K: Serializer, V: Serializer](
+        target: Properties): Unit = {
+      target.put(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG,
+                 Serializer[K].instance.getClass.getName)
+      target.put(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG,
+                 Serializer[V].instance.getClass.getName)
+    }
+  }
+
+  class Common(
```

Comment:
Hm, we should extract out a CommonConfig, and have a ConsumerConfig & ProducerConfig? This file is starting to look a little long. But I could also see wanting to have all the config in one spot.

Reply:
Oh I take that back since I guess we want both of those case classes to be nested inside of the Config Object.

Reply:
It's not a requirement for them to be inside the `Config` object. We can definitely split them out if we think it would be better.

Reply:
Looking back at this I think it would be fine either way. I maybe have a slight preference for splitting them out so it's easier to search for?

Reply:
I split this file up into a `config` package.

I also did some more documentation updates and added a test for the `Producer` class.